site/src/utils/workspace.tsx

```diff
@@ -286,6 +286,12 @@ export const paramsUsedToCreateWorkspace = (
 	param: TypesGen.TemplateVersionParameter,
 ) => !param.ephemeral;
 
+export const getAgents = (workspace: TypesGen.Workspace) => {
+	return workspace.latest_build.resources
+		.flatMap((r) => r.agents)
+		.filter((a) => !!a);
```

Comment:
- explicit boolean cast is clearer than `!!` imo
- doing the `filter` first is more efficient, prevents allocating array space for `undefined` entries

```suggestion
		.filter(Boolean)
		.flatMap((r) => r.agents);
```

Reply:
- `r.agents` can return `undefined`, which is why the filter happens afterward.
- Instead of using `Boolean`, maybe `a !== undefined` would be clearer in this case. What do you think?

Reply:
also I'm now realizing this is another thing getting moved to utils/ that could be better off somewhere else. maybe modules/workspaces/agents.ts?

Reply:
I made a small change to the code so there are no new additions to the workspace utils haha.